### ISSUE TITLE
Feedback Issue: Gitpod Public API 

### ISSUE THREAD

**Issue by loujaybee** [gitpod-io/gitpod#15433]
### Objective

Issue for feedback related to our upcoming Public API. 

**Related roadmap items:** 

- https://github.com/gitpod-io/gitpod/issues/10798
- https://github.com/gitpod-io/gitpod/issues/14280
- https://github.com/gitpod-io/gitpod/issues/10427
- https://github.com/gitpod-io/gitpod/issues/10799
- https://github.com/gitpod-io/gitpod/issues/11335
- https://github.com/gitpod-io/gitpod/issues/15213
- https://github.com/gitpod-io/gitpod/issues/15419

**Comment by mbrevoort** [comment]
Some feedback annoted:
<img width="1049" alt="Screenshot 2022-12-20 at 8 51 41 AM" src="https://user-images.githubusercontent.com/30455/208711293-5d1a91f9-36d2-4f7a-be7b-99f99ddb3044.png">

It useless to offer an API token with "no access". We shouldn't allow it. Consider providing an `identity` scope and basic API method to return information about the identity of the token including the tokens ID and name and scopes. Because once you copy it, there isn't another way to correlate the token you have back to the list of tokens in the UI. If I had 10 tokens and one was compromised, but all I had was the value, how do I know which one to invalidate?

If we are going to have names for tokens, we shouldn't allow duplicates. 
<img width="496" alt="image" src="https://user-images.githubusercontent.com/30455/208712765-fb1cf473-23d1-4ecd-97c8-2770271c509f.png">

**Comment by mbrevoort** [comment]
Additional annotated comments:
<img width="642" alt="Screenshot 2022-12-20 at 9 12 01 AM" src="https://user-images.githubusercontent.com/30455/208714099-f03c6d93-7c63-4350-8997-ba80f360b4b7.png">

**Comment by mbrevoort** [comment]
This seems daftly redundant. How many times do we need to say "access token"? 🤪
<img width="956" alt="Screenshot 2022-12-20 at 9 16 53 AM" src="https://user-images.githubusercontent.com/30455/208714580-85b2854b-cfd5-49fb-a846-267e6a311d80.png">

**Comment by mbrevoort** [comment]
The empty state is an opportunity to educate users about the API so don't waste the space. What can I do with the API? Why would I want to? Why would I want to create an access token? If I'm less savvy and don't know what access tokens are, how do I learn? 
<img width="921" alt="image" src="https://user-images.githubusercontent.com/30455/208714730-beb8d200-3557-46d4-8c85-c21605564a2f.png">

**Comment by Siddhant-K-code** [comment]
Thanks @mbrevoort for the detailed feedback ✨

> consider for relabelling API tokens

I don't think so. I think, PAT Tokens are the superset & API Integrations is a subset of it. Maybe, in future we could come with other various things to do with PAT tokens also.

> When do we planned to another option

Not sure, but for sure. 🔜 (lot of edge cases we could cover & provide some various options to the user)

> Full Access - should be more specific

Yeah, we could describe more in the description text about the permissions user would be providing under Read/Write.

> Should generate a default name

Agreed 👍🏽 

> cURL command to copy & paste to test the token

Since, it is not an REST API, we need to introduce something like [`grpcurl`](https://github.com/fullstorydev/grpcurl) to users to do so.

> This seems daftly redundant. How many times do we need to say "access token"? 🤪

I don't think, is there any workaround for not doing so?

> empty state is an opportunity to educate users about the API

Agreed! We Could Introduce a simple `Learn More` button/link, linking to our docs of Access token

**Comment by ghuntley** [comment]
As a general rule of thumb: the design of tokens should be matchable via Regex expression and include a checksum that can external parties can use to validate offline that the token was issued by Gitpod.

**Comment by Siddhant-K-code** [comment]
> @ghuntley As a general rule of thumb: the design of tokens should be matchable via Regex expression and include a checksum that external parties can use to validate offline that the token was issued by Gitpod.

Yeah, our API PAT Tokens contain `gitpod_pat_` as a starting slug. They can match this pattern for any token validation.
